### PR TITLE
feat: Test Admin Resolver

### LINKED ISSUE #955: 💥 [DevOps] Enforce 60% coverage (backend)
## 💥 DevOps ticket

Enforce 60% coverage (backend)

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest
Test Admin Resolver

### Issues
- fixes #955 

### Todo

- [X] None

### CODE REVIEW

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 1]
```diff
@@ -0,0 +1,1054 @@
...
+              variables.email = 'bibi@bloxberg.de'
+            })
+
+            describe('date of creation is not a date string', () => {
+              it('throws an error', async () => {
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [new GraphQLError('No Creation found!')],
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:555]: This seems to be the wrong error/too unspecific

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 2]
```diff
@@ -0,0 +1,1054 @@
...
+                variables.creationDate = new Date(
+                  now.getFullYear(),
+                  now.getMonth() - 4,
+                  1,
+                ).toString()
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [new GraphQLError('No Creation found!')],
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:573]: This seems to be the wrong error/too unspecific

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 3]
```diff
@@ -0,0 +1,1054 @@
...
+                variables.creationDate = new Date(
+                  now.getFullYear(),
+                  now.getMonth() + 4,
+                  1,
+                ).toString()
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [new GraphQLError('No Creation found!')],
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:591]: This seems to be the wrong error/too unspecific

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 4]
```diff
@@ -0,0 +1,1054 @@
...
+            describe('second creation surpasses the available amount ', () => {
+              it('returns an array of the open creations for the last three months', async () => {
+                variables.amount = new Decimal(1000)
+                await expect(
+                  mutate({ mutation: createPendingCreation, variables }),
+                ).resolves.toEqual(
+                  expect.objectContaining({
+                    errors: [
+                      new GraphQLError(
+                        'The amount (1000 GDD) to be created exceeds the available amount (800 GDD) for this month.',
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:638]:
```suggestion
                        'The amount (1000 GDD) to be created exceeds the amount (800 GDD) still available for this month.',
```

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 5]
```diff
@@ -0,0 +1,1054 @@
...
+          const massCreationVariables = [
+            'bibi@bloxberg.de',
+            'peter@lustig.de',
+            'stephen@hawking.uk',
+            'garrick@ollivander.com',
+            'bob@baumeister.de',
+          ].map((email) => {
+            return {
+              email,
+              amount: new Decimal(500),
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:664]: THis whole construct is bound to fail, what if there is two pending creations for the same amount for the same month for the same user?

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 6]
```diff
@@ -0,0 +1,1054 @@
...
+                }),
+              ).resolves.toEqual(
+                expect.objectContaining({
+                  errors: [new GraphQLError('Moderator can not confirm own pending creation')],
+                }),
+              )
+            })
+          })
+
+          describe('creation of other user', () => {
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:1013]:
```suggestion
          describe('confirm creation for other user', () => {
```

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 7]
```diff
@@ -0,0 +1,1054 @@
...
+              await expect(
+                mutate({
+                  mutation: confirmPendingCreation,
+                  variables: {
+                    id: -1,
+                  },
+                }),
+              ).resolves.toEqual(
+                expect.objectContaining({
+                  errors: [new GraphQLError('Creation not found to given id.')],
```
Comment (ulfgebhardt) [inline backend/src/graphql/resolver/AdminResolver.test.ts:980]:
```suggestion
                  errors: [new GraphQLError('Creation not found for given id.')],
```

File: backend/src/seeds/factory/creation.ts [hunk 8]
```diff
@@ -17,27 +17,22 @@ export const nMonthsBefore = (date: Date, months = 1): string => {
...
-      where: { userId: user.id },
-      order: { created: 'DESC' },
-    })
+  const pendingCreation = await AdminPendingCreation.findOneOrFail({
+    where: { userId: user.id, amount: creation.amount },
+    order: { created: 'DESC' },
+  })
 
+  if (creation.confirmed) {
     await mutate({ mutation: confirmPendingCreation, variables: { id: pendingCreation.id } })
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/seeds/factory/creation.ts:36]: ?

File: backend/src/seeds/factory/user.ts [hunk 9]
```diff
@@ -24,13 +24,15 @@ export const userFactory = async (
     })
   }
 
-  if (user.createdAt || user.deletedAt || user.isAdmin) {
-    // get user from database
-    const dbUser = await User.findOneOrFail({ id })
+  // get user from database
+  const dbUser = await User.findOneOrFail({ id })
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/seeds/factory/user.ts:28]: Why? The newly created user cannot be deleted

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 10]
```diff
@@ -0,0 +1,1054 @@
...
+          beforeAll(async () => {
+            user = await userFactory(testEnv, bibiBloxberg)
+          })
+
+          it('returns date string', async () => {
+            const result = await mutate({ mutation: deleteUser, variables: { userId: user.id } })
+            expect(result).toEqual(
+              expect.objectContaining({
+                data: {
+                  deleteUser: expect.any(String),
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/AdminResolver.test.ts:143]: see line 147

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 11]
```diff
@@ -0,0 +1,1054 @@
...
+          beforeAll(async () => {
+            user = await userFactory(testEnv, bibiBloxberg)
+          })
+
+          it('throws an error', async () => {
+            await expect(
+              mutate({ mutation: unDeleteUser, variables: { userId: user.id } }),
+            ).resolves.toEqual(
+              expect.objectContaining({
+                errors: [new GraphQLError('User already deleted')],
```
Comment (Mogge) [inline backend/src/graphql/resolver/AdminResolver.test.ts:239]:
```suggestion
                errors: [new GraphQLError('User already undeleted')],
```

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 12]
```diff
@@ -0,0 +1,1054 @@
...
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/AdminResolver.test.ts:239]: Good catch

File: backend/src/graphql/resolver/AdminResolver.test.ts [hunk 13]
```diff
@@ -0,0 +1,1054 @@
...
+        afterAll(async () => {
+          await cleanDB()
+          resetToken()
+        })
+
+        describe('createPendingCreation', () => {
+          beforeAll(async () => {
+            const now = new Date()
+            creation = await creationFactory(testEnv, {
+              email: 'peter@lustig.de',
```
[the remark this answers is not in the document]
Comment (Mogge) [reply backend/src/graphql/resolver/AdminResolver.test.ts:496]: The creation is done by email. I agree that it would be better by ID

**Pull request merged**